### PR TITLE
feat: add support for Argo Rollouts

### PR DESCRIPTION
**Pull request by Soumil-07**

### CODE REVIEW

File: types/ir/ir.go [hunk 1]
```diff
@@ -113,8 +125,8 @@ type Service struct {
 	Replicas                    int
 	Networks                    []string
 	OnlyIngress                 bool
-	Daemon                      bool //Gets converted to DaemonSet
-	StatefulSet                 bool // Whether the service gets converted to a StatefulSet
+	Daemon                      bool           //Gets converted to DaemonSet
+	DeploymentType              DeploymentType // Whether the service gets converted to a StatefulSet
```
Comment (Akash-Nayak) [inline types/ir/ir.go:129]: Can we change the comment as it's not just for `StatefulSet`?

File: types/qaengine/commonqa/commonqa.go [hunk 2]
```diff
@@ -152,8 +153,20 @@ func GetPortForService(detectedPorts []int32, qaSubKey string) int32 {
 	return int32(selectedPort)
 }
 
-// IsStateful returns whether the Service should generate a StatefulSet
-func IsStateful(serviceName string) bool {
-	quesKey := common.JoinQASubKeys(common.ConfigServicesKey, serviceName, common.ConfigStatefulSetKey)
-	return qaengine.FetchBoolAnswer(quesKey, fmt.Sprintf("For the service '%s', do you require a StatefulSet instead of a Deployment?", serviceName), nil, false, nil)
+// GetDeploymentType returns whether the Service should generate a StatefulSet
```
Comment (Akash-Nayak) [inline types/qaengine/commonqa/commonqa.go:156]: Can we update this comment too?

**Review by Akash-Nayak — CHANGES_REQUESTED** [review]

File: types/qaengine/commonqa/commonqa.go (hunk 2 above)
Comment (Soumil-07) [reply types/qaengine/commonqa/commonqa.go]: Resolved in [1608c10](https://github.com/konveyor/move2kube/pull/1098/commits/1608c1008a61017cf59810042d9021b06b034ae7)

File: types/ir/ir.go (hunk 1 above)
Comment (Soumil-07) [reply types/ir/ir.go]: Resolved in [1608c10](https://github.com/konveyor/move2kube/pull/1098/commits/1608c1008a61017cf59810042d9021b06b034ae7)

File: go.mod [hunk 3]
```diff
@@ -147,129 +151,135 @@ require (
...
 	gopkg.in/inf.v0 v0.9.1 // indirect
-	gopkg.in/ini.v1 v1.66.2 // indirect
+	gopkg.in/ini.v1 v1.67.0 // indirect
 	gopkg.in/warnings.v0 v0.1.2 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
-	k8s.io/apiextensions-apiserver v0.23.4 // indirect
+	k8s.io/apiextensions-apiserver v0.25.8 // indirect
 	k8s.io/apiserver v0.26.2 // indirect
-	k8s.io/cli-runtime v0.23.1 // indirect
+	k8s.io/cli-runtime v0.25.8 // indirect
```
[the remark this answers is not in the document]
Comment (Soumil-07) [reply go.mod:233]: If I'm not wrong, these are indirect dependencies so they have to 25.8

File: common/utils.go [hunk 4]
```diff
@@ -1159,6 +1159,7 @@ func MarshalObjToYaml(obj runtime.Object) ([]byte, error) {
 		logrus.Errorf("Error while marshalling object %+v to json. Error: %q", obj, err)
 		return nil, err
 	}
+	fmt.Printf("json: %s\n", string(objJSONBytes))
```
[the remark this answers is not in the document]
Comment (Soumil-07) [reply common/utils.go]: Resolved in [1608c10](https://github.com/konveyor/move2kube/pull/1098/commits/1608c1008a61017cf59810042d9021b06b034ae7)

**Review by Akash-Nayak — APPROVED** [review]
LGTM

**Review by HarikrishnanBalagopal — APPROVED** [review]
LGTM

**Pull request merged**